### PR TITLE
Fix flaky checks

### PR DESCRIPTION
**Pull request by azat**
CI [1]:

    2024-03-02T22:23:37.4878824Z Going to start run command [python3 /home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/ci/functional_test_check.py "$CHECK_NAME" $KILL_TIMEOUT]
    2024-03-02T22:23:37.7538256Z Runnin check [Stateless tests flaky check (asan)] with timeout [3600]
    2024-03-02T22:23:37.8790465Z Fetched info about 3 changed files
    2024-03-02T22:23:37.9160414Z Traceback (most recent call last):
    2024-03-02T22:23:37.9161715Z   File "/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/ci/functional_test_check.py", line 341, in <module>
    2024-03-02T22:23:37.9162714Z     main()
    2024-03-02T22:23:37.9163683Z   File "/home/ubuntu/actions-runner/_work/ClickHouse/ClickHouse/tests/ci/functional_test_check.py", line 235, in main
    2024-03-02T22:23:37.9164642Z     assert (
    2024-03-02T22:23:37.9165273Z AssertionError: JobReport file path must be provided with --validate-bugfix
    2024-03-02T22:23:37.9612679Z Run action failed for: [Stateless tests flaky check (asan)] with exit code [1]

  [1]: https://github.com/ClickHouse/ClickHouse/actions/runs/8125551797/job/22208766466?pr=60689

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow-up for: #59348 (cc @maxknv)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 418a6258ac1b55d5e225107bd9ebfc711ca6511b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60694/418a6258ac1b55d5e225107bd9ebfc711ca6511b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by azat** [comment]
>Upgrade check (debug) — check_status.tsv doesn't exists


```
2024.03.03 11:24:39.766495 [ 22325 ] {} <Error> Application: Code: 210. DB::Exception: Listen [::]:8123 failed: Code: 318. DB::Exception: Unknown handler type 'redirect' in config here: http_handlers.rule.handler.type. (INVALID_CONFIG_PARAMETER) (version 24.2.1.2248 (official build)). (NETWORK_ERROR), Stack trace (when copying this message, always include the lines below):
```

File: tests/ci/functional_test_check.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def main():
     run_changed_tests = flaky_check or validate_bugfix_check
     pr_info = PRInfo(need_changed_files=run_changed_tests)
     tests_to_run = []
-    if run_changed_tests:
+    if validate_bugfix_check:
         assert (
             args.report_to_file
         ), "JobReport file path must be provided with --validate-bugfix"
```
Comment (maxknv) [inline tests/ci/functional_test_check.py:237]:
```suggestion
    assert (
        not validate_bugfix_check or args.report_to_file
    ), "JobReport file path must be provided with --validate-bugfix"
    if run_changed_tests:
```

Comment (maxknv) [reply tests/ci/functional_test_check.py]: Thanks for bringing this up.
We still need fetch list of tests in `if` for both "validate-bugfix" and "flaky-check", so I propose removing assert from this `if` to fix the bug

**Review by maxknv — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**